### PR TITLE
containers: Add info for Azure Container Registry

### PR DESCRIPTION
**Pull request by doanac**
Signed-off-by: Andy Doan <andy@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
@kprosise @mike-sul - once we merge https://github.com/foundriesio/ci-scripts/pull/169 we'll need this doc update

**Comment by doanac** [comment]
Docs for f99e12527c3cc0b7dcdfd0210e2a86d29cfc42cf are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1287/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 97dc02063f0590e4c461a198aa659bb8c8c88602 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1288/docs/artifacts/html/index.html

**Review by mike-sul — APPROVED** [review]

File: source/reference-manual/docker/private-registries.rst [hunk 1]
```diff
@@ -49,3 +50,25 @@ Then set the factory configuration with something like::
 
 .. _aws ecr get-login-password:
    https://docs.aws.amazon.com/cli/latest/reference/ecr/get-login-password.html
+
```
Comment (angolini) [inline source/reference-manual/docker/private-registries.rst:53]: additional empty line (only a reminder, I don't know if this is intentional)

**Review by angolini — APPROVED** [review]
LGTM

File: source/reference-manual/docker/private-registries.rst [hunk 2]
```diff
@@ -49,3 +50,25 @@ Then set the factory configuration with something like::
...
+---------------------------------------------
+
+CI can be configured to use an ACR `service principal`_ with read-only
+access to a private ACR instance. First, CI must be configured with
+the service principal's ID and password::
+
+ $ fioctl secrets update azprincipal='<ID>:<PASSWORD>'
+
+The factory :ref:`configuration <ref-factory-definition>` is then
+updated with something like::
```
Comment (kprosise) [inline source/reference-manual/docker/private-registries.rst:65]:
```suggestion
updated accordingly::
```

**Review by kprosise — APPROVED** [review]
Suggestions provided, though totally optional.

**Comment by doanac** [comment]
Docs for 8b20489347391788aee380acc32c18c0ec4d4c4f are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1293/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 8b20489347391788aee380acc32c18c0ec4d4c4f are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1294/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 2302496f9bc26c3370258ba0d13526b096ccd57c are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1292/docs/artifacts/html/index.html

**Comment by doanac** [comment]
rebase and pushed

**Comment by doanac** [comment]
Docs for febb000e9b04af431c8ca9dc1939abbf058f15b4 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1295/docs/artifacts/html/index.html

**Pull request merged**